src/analysis_and_optimization/Partial_evaluator.ml

```diff
@@ -629,12 +674,12 @@ let rec eval_expr (e : Middle.expr_typed_located) =
                 (* Constant folding for operators *)
             | op, [{expr= Lit (Int, i); _}] -> (
               match op with
-              | "PPlus_" | "PMinus__" | "PNot__" ->
+              | "PPlus__" | "PMinus__" | "PNot__" ->
                   apply_prefix_operator_int op (Int.of_string i)
               | _ -> FunApp (StanLib, op, l) )
             | op, [{expr= Lit (Real, r); _}] -> (
               match op with
-              | "PPlus_" | "PMinus__" ->
+              | "PPlus__" | "PMinus__" ->
```

Comment:
I think ideally we want to try to use the pattern Micky gave us, which is to match on any string and put a `when` guard checking which kind it is using `internal_fn_of_string` or `operator_of_string`.